acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/steps/ContentSteps.java

```diff
@@ -102,52 +114,100 @@ public void whenIPublishThePublication() throws Throwable {
+                consumableAttachmentElement.getSizeText(), is("size: " + attachment.getContent().length + " B."));
 
-        assertThat("Granularity is as expected",consumablePublicationPage.getGranularity(),
-            is("Granularity:\n" + publication.getGranularity()));
+
+            if (acceptanceTestProperties.isHeadlessMode()) {
+                // At the moment of writing, there doesn't seem to be any easy way available to force Chromedriver
+                // to download files when operating in headless mode. It appears that some functionality has been
+                // added to DevTools but it's not obvious how to trigger that from Java so, for now at least,
+                // we'll only be testing file download when operating in a full, graphical mode.
+                //
+                // See bug report at https://bugs.chromium.org/p/chromium/issues/detail?id=696481 and other reports
+                // available online.
+                log.warn("Not testing file download due to running in a headless mode.");
+            } else {
+
+                // Trigger file download by click the <a> tag.
+                consumableAttachmentElement.clickHyperlink();
+
+                final Path downloadedFilePath = Paths.get(acceptanceTestProperties.getDownloadDir().toString(),
+                    attachment.getFullName());
+
+                waitUntilFileAppears(downloadedFilePath);
+
+                assertThat("Downloaded file has the same content as the uploaded attachment "
+                        + attachment.getFullName(),
+                    readFileAsByteArray(downloadedFilePath), is(attachment.getContent())
+                );
+            }
+        });
+    }
+
+    @When("^I try to upload a file of one of the forbidden types:$")
+    public void iTryToUploadAFileOfOneOfTheForbiddenTypes(final DataTable forbiddenExtensions) throws Throwable {
+
+        final List<Attachment> forbiddenAttachments = forbiddenExtensions.asList(String.class).stream()
+            .map(extension -> FileType.valueOf(extension.toUpperCase()))
+            .map(forbiddenFileType -> AttachmentBuilder
+                .createAttachment()
+                .withFileType(forbiddenFileType)
+                .build())
+            .collect(toList());
+
+        testDataRepo.setCurrentAttachments(forbiddenAttachments);
+    }
+
+    @Then("^the upload is rejected with an error message$")
+    public void theUploadIsRejectedWithAnErrorMessage() throws Throwable {
+
+        testDataRepo.getCurrentAttachments().forEach(attachment -> {
+            contentPage.getAttachmentsSection().uploadAttachment(attachment);
+
+            final String expectedErrorMessage = MessageFormat.format(
+                "The uploaded file ''{0}'' has extension {1} which is not allowed." +
+                    " Allowed extensions are: doc, docx, xls, xlsx, pdf, csv, zip, txt, rar, ppt, pptx.",
```

Comment:
Suggest we take this list of extensions out, since this may get out of date, and I don't think it's necessary anyway if we can match on what the string *starts with*

Reply:
Good point. The static list is now removed and the matcher switched to use 'startsWith`.